File: benchmark/fibonacci/fibonacci.cpp
#include <iostream>
#include <time.h>

long fib(int n){
  if(n == 0)return 0;
  if(n == 1)return 1;
  return fib(n-1) + fib(n-2);
}

int main(){

  int N = 40;
  clock_t start = clock();
  long fibonacci = fib(N);
  clock_t end = clock();

  std::cout << "C++ " << fibonacci << " " << (double)(end-start)/CLOCKS_PER_SEC << "sec." << std::endl;

  return 0;
}

File: benchmark/fibonacci/fibonacci.rs
use std::time::Instant;

fn main(){
    let n = 40;
    let start = Instant::now();
    let fib = fibn(n);
    let end = start.elapsed();
    println!("rust {} {}sec.", fib, (end.as_secs() as f64) + (end.subsec_nanos() as f64) / 1000_000_000.);
}

fn fibn(n: u64) -> u64 {
    if n <= 1 {
        n
    } else {
        fibn(n-1) + fibn(n-2)
    }
}
